scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1485,6 +1495,17 @@ def eigsh(A, k=6, M=None, sigma=None, which='LM', v0=None,
     (13, 6)
 
     """
+    n = A.shape[0]
+
+    if k >= n:
+        import warnings
+        warnings.warn('k greater than/equal to the order of the square '
+                      'matrix. Using scipy.linalg.eig instead.')
```

Comment:
This should use a `SparseEfficiencyWarning` rather than a generic `UserWarning`.